### PR TITLE
1.4 release notes.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Review by RAMitchell — APPROVED** [review]

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,211 @@ XGBoost Change Log
...
+In 1.4 we overhauled the underlying prediction functions for C API and Python API with an
+unified interface. (#6777, #6693, #6653, #6662, #6648, #6668)
+* Starting with 1.4, sklearn interface prediction will use inplace predict by default when
+  input data is supported.
+* Users can use inplace predict with ``dart`` booster and enable GPU acceleration just
+  like ``gbtree``.
+* Also all prediction functions with tree models are now thread-safe.  Inplace predict is
+  improved with ``base_margin`` support.
+* A new set of C predict functions are exposed in the public interface.
+* A user-visible change is a newly added parameter called ``stric_shape``.  See
```
Comment (hcho3) [inline NEWS.md:19]:
```suggestion
* A user-visible change is a newly added parameter called ``strict_shape``.  See
```

File: NEWS.md [hunk 2]
```diff
@@ -3,6 +3,211 @@ XGBoost Change Log
...
+  https://xgboost.readthedocs.io/en/latest/prediction.html for more details.
+
+
+### Improvement on Dask interface
+* Starting with 1.4, the dask interface is considered to be feature complete, which means
```
Comment (hcho3) [inline NEWS.md:24]:
```suggestion
* Starting with 1.4, the Dask interface is considered to be feature-complete, which means
```

Comment (hcho3) [reply NEWS.md:24]: In general, please capitalize the first letter: Dask

File: NEWS.md [hunk 3]
```diff
@@ -3,6 +3,211 @@ XGBoost Change Log
...
+  - Evaluation result is printed on the scheduler process. (#6609)
+  - ``DaskDMatrix`` (and device quantile dmatrix) now accepts all meta-information. (#6601)
+
+* Prediction optimization.  We enhanced and speeded up the prediction function for the
+  dask interface.  See the latest dask tutorial page in our document for an overview of
+  how you can optimize it even further. (#6650, #6645, #6648, #6668)
+
+* Bug fixes
+  - If you are using the latest dask and distributed where ``distributed.MultiLock`` is
+    presented, XGBoost supports training multiple models on the same cluster in
```
Comment (hcho3) [inline NEWS.md:45]:
```suggestion
    present, XGBoost supports training multiple models on the same cluster in
```

File: NEWS.md [hunk 4]
```diff
@@ -3,6 +3,211 @@ XGBoost Change Log
...
+  - Shap value computation is now supported. (#6575, #6645, #6614)
+  - Evaluation result is printed on the scheduler process. (#6609)
+  - ``DaskDMatrix`` (and device quantile dmatrix) now accepts all meta-information. (#6601)
+
+* Prediction optimization.  We enhanced and speeded up the prediction function for the
+  dask interface.  See the latest dask tutorial page in our document for an overview of
+  how you can optimize it even further. (#6650, #6645, #6648, #6668)
+
+* Bug fixes
+  - If you are using the latest dask and distributed where ``distributed.MultiLock`` is
```
Comment (hcho3) [inline NEWS.md:44]:
```suggestion
  - If you are using the latest `dask` and `distributed` where `distributed.MultiLock` is
```

File: NEWS.md [hunk 5]
```diff
@@ -3,6 +3,211 @@ XGBoost Change Log
...
+    presented, XGBoost supports training multiple models on the same cluster in
+    parallel. (#6743)
+  - A bug fix for when using ``dask.client`` to launch async task, XGBoost might use a
```
Comment (hcho3) [inline NEWS.md:47]:
```suggestion
  - A bug fix for when using `dask.client` to launch async task, XGBoost might use a
```

File: NEWS.md [hunk 6]
```diff
@@ -3,6 +3,211 @@ XGBoost Change Log
 
 This file records the changes in xgboost library in reverse chronological order.
 
+## v1.4.0 (2021.04.12)
+
```
Comment (hcho3) [inline NEWS.md:7]:
```suggestion
### Introduction of pre-built binary package for R, with GPU support
Starting with release 1.4.0, users now have the option of installing `{xgboost}` without having to build it from the source. This is particularly advantageous for users who want to take advantage of the GPU algorithm (`gpu_hist`), as previously they'd have to build `{xgboost}` from the source using CMake and NVCC. Now installing `{xgboost}` with GPU support is as easy as:
`R CMD INSTALL ./xgboost_r_gpu_linux.tar.gz`

See the instructions at https://xgboost.readthedocs.io/en/latest/build.html
```

**Review by hcho3 — CHANGES_REQUESTED** [review]
A few backported PRs appear to be left out:  #6797, #6817, #6826, #6827. Also, we should highlight the introduction of the binary package for R.

File: NEWS.md (hunk 6 above)
Comment (trivialfis) [reply NEWS.md:7]: @hcho3 Thanks for reminding me, why use `{xgboost}`, specifically the `{}`?

Comment (hcho3) [reply NEWS.md:7]: This is the convention used in the R user community to indicate the name of R packages. Example: `{caret}`

Comment (trivialfis) [reply NEWS.md:7]: Got it, thanks for the explanation.

**Comment by trivialfis** [comment]
@hcho3 All comments are resolved.

**Review by hcho3 — APPROVED** [review]

**Review by hcho3 — CHANGES_REQUESTED** [review]
Please also add the list of deprecations and known issues listed in https://github.com/dmlc/xgboost/issues/6793#issuecomment-809511757

**Comment by trivialfis** [comment]
@hcho3 Done.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**